File: CodingSamples/DotNet/Runtime/ReflectionTest2/FinLib/HomeLoan.cs
using System.Runtime.CompilerServices;

namespace Finance;


public class HomeLoan
{
    public float Common(double amount, int period) => period < 5 ? 9.0f : 8.5f;

    [MaxDuration(12)] //custom attribute
    public float Woman(double amount, int period) => Common(amount, period) - 1;

    [MethodImpl(MethodImplOptions.Synchronized)] //pseudo attribute
    public float Soldier(double amount, int period) => Common(amount, period) - 4;

}

File: CodingSamples/Java/Runtime/ReflectionTest2/src/HomeLoan.java
package finance;

public class HomeLoan {

	public float common(double amount, int period) {
		return period < 5 ? 9.0f : 8.5f;
	}

	@MaxDuration(value=12)
	public float woman(double amount, int period) {
		return common(amount, period) - 1;
	}

	public float soldier(double amount, int period) {
		return common(amount, period) - 4;
	}
}

